File: SURVEYllance/ClientApp/webroot/src/sharedObjects/SurveyAnswer.js
/**
 * answers for a Survey
 *  @property {string} Id id to identify the answer from backend calls
 *  @property {string} Text text of the answer
 *  @property {number} Votes Number of votes for this answer
 */
class SurveyAnswer {

    //<editor-fold desc="Private properties">

    /**
     * id to identify the answer from backend calls
     * @type {string}
     * @private
     */
    #id;

    /**
     * text of the answer
     * @type {string}
     * @private
     */
    #text;

    /**
     * Number of votes. Can only be incremented (ath the moment) since votes can't be taken back
     * @type {number}
     * @private
     */
    #votes;

    //</editor-fold>

    //<editor-fold desc="Getter and Setter">

    /**
     * Getter of {@link #id}
     * @returns {string}
     */
    get id() {
        return this.#id;
    }

    /**
     * Getter of {@link #text}
     * @returns {string}
     */
    get text() {
        return this.#text;
    }

    /**
     * Getter of {@link #votes}
     * @returns {number}
     */
    get votes() {
        return this.#votes;
    }

    /**
     * Setter of {@link #votes}
     * Check if votes have been increased
     * @param {string} value new value
     */
    set votes(value) {
        if (this.#votes === undefined || this.#votes < value) {
            this.#votes = value;
            //TODO: fire event?
        }
    }

    //</editor-fold>

    //<editor-fold desc="Constructor">

    /**
     * Constructor to create answers
     * @param {string} text Name of the answer. NOT the number displayed before, which should be the Index
     * @constructor
     */
    constructor(text){
        this.#text = text;
    }

    //</editor-fold>

    /**
     * This method overrides the default toJSON method, and returns object WITH the private properties, so they will be passed to the Backend
     * @return {{votes: number, id: string, text: string}} Object with the private properties
     * @override
     */
    toJSON() {
        return {
            id: this.#id,
            text: this.#text,
            votes: this.#votes
        }
    }
}

File: SURVEYllance/Resources/SurveyAnswer.cs
using System;

namespace SURVEYllance.Resources
{
    /// <summary>
    /// Answers for a <see cref="Survey"/>
    /// </summary>
    public class SurveyAnswer
    {
        #region Private properties
        
        /// <summary>
        /// Id to identify the answer from frontend calls
        /// </summary>
        private readonly string _id;

        /// <summary>
        /// The text of the Answer. Possible answer for <see cref="Survey"/>
        /// </summary>
        private readonly string _text;
        
        /// <summary>
        /// Number of votes. Can only be incremented (ath the moment) since votes can't be taken back
        /// </summary>
        private int _votes = 0;

        #endregion

        #region Getter and Setter
        
        /// <summary>
        /// Getter for <see cref="_id"/>
        /// Id is generated on construction
        /// </summary>
        public string Id => _id;

        /// <summary>
        /// Getter for <see cref="_text"/>
        /// The Text of an answer can only be read and will be set on construction
        /// </summary>
        public string Text => _text;

        /// <summary>
        /// Getter and Setter for <see cref="_votes"/>
        /// Check if votes have been increased. If so fire <see cref="OnVotesChange"/>-event
        /// </summary>
        public int Votes
        {
            get => _votes;
            set
            {
                if (value <= _votes) return;
                _votes = value;
                OnVotesChange?.Invoke();
            }
        }

        #endregion

        #region Constructor

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="text">Text of the Answer. Possible Answer for <see cref="Survey"/></param>
        public SurveyAnswer(string text)
        {
            _id = Convert.ToBase64String(Guid.NewGuid().ToByteArray());
            _text = text;
        }

        #endregion

        #region Event handling

        // Delegate type for the event handler
        /// <summary>
        /// Delegate of <see cref="SurveyAnswer.OnVotesChange"/>
        /// </summary>
        public delegate void VotesChange();

        // Declare the event.
        /// <summary>
        /// Fire when the number of votes for this Answer increases
        /// </summary>
        public event VotesChange OnVotesChange;

        #endregion
        
    }
}